cmake/kconfig.cmake

```diff
@@ -104,6 +104,38 @@ if(EXTRA_KCONFIG_OPTIONS)
     )
 endif()
 
+# Set overlay config files list using a list of directories where the
+# overlay files could be found.
+if(DEFINED ENV{OVERLAY_CONFIG_DIRS})
+  set(OVERLAY_CONFIG_DIRS $ENV{OVERLAY_CONFIG_DIRS})
+
+  # We add current directory first so that it is seached first
+  set(overlay_config_dir_list ".:${OVERLAY_CONFIG_DIRS}")
+  string(REPLACE ":" ";" overlay_config_dirs "${overlay_config_dir_list}")
```

Comment:
: is used in Windows (C:), use '|' instead.

Reply:
Or maybe ' ' ?

Reply:
I was following the way directories are separated in PATH in unix.

Reply:
Right, but we need to also support Windows.

Reply:
Sure, we should just figure out what separator to use. Probably not space as that could be part of file or directory. Perhaps ";" could be used?

Reply:
I'd prefer to use what we are already using, and that is whitespace.

https://github.com/zephyrproject-rtos/zephyr/pull/15669/files/f27c6b30e2229c3710b242482ae89e1904f53b85#diff-3a17f68c7317a0565fa6d10ecdd67c05R26

We can't have every user-settable list use a different separator.